docs/source/notes/get_start_xpu.rst

```diff
@@ -0,0 +1,417 @@
+   cd pytorch
+   git checkout main # or checkout the specific release version >= v2.4
+   git submodule sync
+   git submodule update --init --recursive
+
+   # Get required packages for compilation
+   conda install cmake ninja
+   pip install -r requirements.txt
+
+   # Pytorch for Intel GPUs only support Linux platform for now.
+   # Install the required packages for pytorch compilation
+   conda install intel::mkl-static intel::mkl-include
+
+   # (optional) If using torch.compile with inductor/triton, install the matching version of triton
+   # Run from the pytorch directory after cloning
+   # For Intel GPU support, please explicitly `export USE_XPU=1` before running command.
+   make triton
+
+   # If you would like to compile PyTorch with new C++ ABI enabled, then first run this command:
+   export _GLIBCXX_USE_CXX11_ABI=1
+
+   # pytorch build from source
+   export CMAKE_PREFIX_PATH=${CONDA_PREFIX:-"$(dirname $(which conda))/../"}
+   python setup.py develop
+   cd ..
+
+
+   # (optional) If using torchvison.
+   # Get torchvision Code
+   git clone https://github.com/pytorch/vision.git
+   cd vision
+   git checkout main # or specific version
+   python setup.py develop
+   cd ..
+
+   # (optional) If using torchaudio.
+   # Get torchaudio Code
+   git clone https://github.com/pytorch/audio.git
+   cd audio
+   pip install -r requirements.txt
+   git checkout main # or specific version
+   git submodule sync
+   git submodule update --init --recursive
+   python setup.py develop
+   cd ..
+
+Check for Intel GPU
+-------------------
+
+**Note**\ : Make sure the environment is properly set up by following `Environment Set up Session <#set-up-environment>`_ before running the code.
+
+To check if your Intel GPU is available, you would typically use the following code:
+
+.. code-block::
+
+   import torch
+   torch.xpu.is_available()  # torch.xpu is the API for Intel GPU support
+
+If the output is ``False``\ , please ensure that you have Intel GPU in your system and correctly follow the `PyTorch Installation Prerequisites for Intel GPUs <https://www.intel.com/content/www/us/en/developer/articles/tool/pytorch-prerequisites-for-intel-gpus.html>`_ and the pytorch compilation is correctly finished.
+
+Minimum Code Change
+-------------------
+
+If you are migrating code from ``cuda``\ , you would change references from ``cuda`` to ``xpu``\ :
+
+For example:
+
+.. code-block::
+
+   # CUDA CODE
+   tensor = torch.tensor([1.0, 2.0]).to("cuda")
+
+   # CODE for Intel GPU
+   tensor = torch.tensor([1.0, 2.0]).to("xpu")
+
+The following points outline the support and limitations for PyTorch with Intel GPU:
+
+
+#. Both Training and inference workflows are supported.
+#. Both eager mode and ``torch.compile`` is supported.
+#. Data types such as FP32, BF16, FP16, and Automatic Mixed Precision (AMP) are all supported.
+#. Models that depend on third-party components, will not be supported until PyTorch v2.5 or later.
```

Comment:
* Autocast was not supported last I tried. Is it now?
* Memory pinning not supported. https://github.com/pytorch/pytorch/issues/126491
* Memory operations under torch.xpu not supported. https://github.com/pytorch/pytorch/issues/127929

Reply:
autocast is now supported, but for train with GradScaler in FP16, there are still 2 ops under implemented.

Reply:
> autocast is now supported

Good to know. I need to give it a try.